src/Features/CSharp/Portable/StringIndentation/CSharpStringIndentationService.cs

```diff
@@ -33,59 +32,52 @@ public CSharpStringIndentationService()
+            nodeStack.Add(root);
+
+            while (nodeStack.TryPop(out var node))
             {
-                var (node, nextChild) = nextNode;
+                // Dont' bother recursing into nodes that don't hit the requested span, they can never contribute 
+                // regions of interest.
+                if (!node.Span.IntersectsWith(textSpan))
+                    continue;
 
-                if (nextChild == 0)
+                if (node is InterpolatedStringExpressionSyntax interpolatedString &&
+                    interpolatedString.StringStartToken.IsKind(SyntaxKind.InterpolatedMultiLineRawStringStartToken))
                 {
-                    // This is the first time we are processing this node
-                    if (!node.Span.IntersectsWith(textSpan))
-                        continue;
-
-                    if (node is InterpolatedStringExpressionSyntax interpolatedString &&
-                        interpolatedString.StringStartToken.IsKind(SyntaxKind.InterpolatedMultiLineRawStringStartToken))
-                    {
-                        ProcessInterpolatedStringExpression(text, interpolatedString, result, cancellationToken);
-                    }
+                    ProcessInterpolatedStringExpression(text, interpolatedString, result, cancellationToken);
                 }
 
-                var childList = node.ChildNodesAndTokens();
-                for (var i = nextChild; i < childList.Count; i++)
+                foreach (var child in node.ChildNodesAndTokens().Reverse())
                 {
-                    var child = childList[i];
                     if (child.IsNode)
                     {
-                        // If necessary, push the current node so we can pick back up after processing the child
-                        if (i < childList.Count - 1)
-                        {
-                            nodeStack.Push((node, i + 1));
-                        }
-
-                        // Push the child for immediate processing
-                        nodeStack.Push((child.AsNode()!, 0));
-
-                        // Break out of the 'for' loop, which effectively continues the containing 'while' loop
-                        break;
+                        nodeStack.Add(child.AsNode()!);
                     }
-                    else if (child.IsKind(SyntaxKind.MultiLineRawStringLiteralToken) ||
-                             child.IsKind(SyntaxKind.Utf8MultiLineRawStringLiteralToken))
+                    else if (child.IsToken)
                     {
-                        ProcessMultiLineRawStringLiteralToken(text, child.AsToken(), result, cancellationToken);
+                        if (child.IsKind(SyntaxKind.MultiLineRawStringLiteralToken) ||
+                            child.IsKind(SyntaxKind.Utf8MultiLineRawStringLiteralToken))
+                        {
+                            ProcessMultiLineRawStringLiteralToken(text, child.AsToken(), result, cancellationToken);
```

Comment:
❓ Are we sure this part isn't a behavior change? It seems like multi-line raw string literals are now going to be inserted into the result list in a reverse BFS order, while the rest of the nodes will be in forward DFS order.

Reply:
in terms of tagging, it doesn't matter.  the tags we produce at this layer are collected higher, then stored in an interval tree (which naturally sorts, and returns the tag-spans in sorted order).